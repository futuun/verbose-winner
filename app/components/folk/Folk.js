import React, { PropTypes } from 'react'
import classNames from 'classnames'
import styles from './styles.sass'

function Folk(props) {
  const { missedLetters } = props
  const flow = [
    'head',
    'neck',
    'corpus',
    'right-arm',
    'left-arm',
    'right-hand',
    'left-hand',
    'right-leg',
    'left-leg',
    'right-foot',
    'left-foot',
  ]

  return (
    <div className={styles.folk}>
      <div className={classNames(styles.img, styles.bar)} />
      {missedLetters.map((val, i) => (
        <div key={i} className={classNames(styles.img, styles[flow[i]])} />
      ))}
    </div>
  )
}

Folk.defaultProps = {
  missedLetters: [],
}

Folk.propTypes = {
  missedLetters: PropTypes.arrayOf(PropTypes.string).isRequired,
}

export default Folk
