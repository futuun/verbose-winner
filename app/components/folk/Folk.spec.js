import React from 'react'
import renderer from 'react-test-renderer'
import Folk from './Folk'

for (let i = 0; i < 12; i++) {
  test(`Folk: ${i} letter${i === 1 ? '' : 's'}`, () => {
    const component = renderer.create(
      <Folk missedLetters={Array(i).fill('x')} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
}
