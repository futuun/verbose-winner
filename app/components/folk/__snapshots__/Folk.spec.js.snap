exports[`test Folk: 0 letters 1`] = `
<div
  className="folk">
  <div
    className="img bar" />
</div>
`;

exports[`test Folk: 1 letter 1`] = `
<div
  className="folk">
  <div
    className="img bar" />
  <div
    className="img head" />
</div>
`;

exports[`test Folk: 2 letters 1`] = `
<div
  className="folk">
  <div
    className="img bar" />
  <div
    className="img head" />
  <div
    className="img neck" />
</div>
`;

exports[`test Folk: 3 letters 1`] = `
<div
  className="folk">
  <div
    className="img bar" />
  <div
    className="img head" />
  <div
    className="img neck" />
  <div
    className="img corpus" />
</div>
`;

exports[`test Folk: 4 letters 1`] = `
<div
  className="folk">
  <div
    className="img bar" />
  <div
    className="img head" />
  <div
    className="img neck" />
  <div
    className="img corpus" />
  <div
    className="img right-arm" />
</div>
`;

exports[`test Folk: 5 letters 1`] = `
<div
  className="folk">
  <div
    className="img bar" />
  <div
    className="img head" />
  <div
    className="img neck" />
  <div
    className="img corpus" />
  <div
    className="img right-arm" />
  <div
    className="img left-arm" />
</div>
`;

exports[`test Folk: 6 letters 1`] = `
<div
  className="folk">
  <div
    className="img bar" />
  <div
    className="img head" />
  <div
    className="img neck" />
  <div
    className="img corpus" />
  <div
    className="img right-arm" />
  <div
    className="img left-arm" />
  <div
    className="img right-hand" />
</div>
`;

exports[`test Folk: 7 letters 1`] = `
<div
  className="folk">
  <div
    className="img bar" />
  <div
    className="img head" />
  <div
    className="img neck" />
  <div
    className="img corpus" />
  <div
    className="img right-arm" />
  <div
    className="img left-arm" />
  <div
    className="img right-hand" />
  <div
    className="img left-hand" />
</div>
`;

exports[`test Folk: 8 letters 1`] = `
<div
  className="folk">
  <div
    className="img bar" />
  <div
    className="img head" />
  <div
    className="img neck" />
  <div
    className="img corpus" />
  <div
    className="img right-arm" />
  <div
    className="img left-arm" />
  <div
    className="img right-hand" />
  <div
    className="img left-hand" />
  <div
    className="img right-leg" />
</div>
`;

exports[`test Folk: 9 letters 1`] = `
<div
  className="folk">
  <div
    className="img bar" />
  <div
    className="img head" />
  <div
    className="img neck" />
  <div
    className="img corpus" />
  <div
    className="img right-arm" />
  <div
    className="img left-arm" />
  <div
    className="img right-hand" />
  <div
    className="img left-hand" />
  <div
    className="img right-leg" />
  <div
    className="img left-leg" />
</div>
`;

exports[`test Folk: 10 letters 1`] = `
<div
  className="folk">
  <div
    className="img bar" />
  <div
    className="img head" />
  <div
    className="img neck" />
  <div
    className="img corpus" />
  <div
    className="img right-arm" />
  <div
    className="img left-arm" />
  <div
    className="img right-hand" />
  <div
    className="img left-hand" />
  <div
    className="img right-leg" />
  <div
    className="img left-leg" />
  <div
    className="img right-foot" />
</div>
`;

exports[`test Folk: 11 letters 1`] = `
<div
  className="folk">
  <div
    className="img bar" />
  <div
    className="img head" />
  <div
    className="img neck" />
  <div
    className="img corpus" />
  <div
    className="img right-arm" />
  <div
    className="img left-arm" />
  <div
    className="img right-hand" />
  <div
    className="img left-hand" />
  <div
    className="img right-leg" />
  <div
    className="img left-leg" />
  <div
    className="img right-foot" />
  <div
    className="img left-foot" />
</div>
`;
